src/main/java/appeng/spatial/StorageCellLot.java

```diff
@@ -0,0 +1,85 @@
+package appeng.spatial;
+
+import net.minecraft.util.math.BlockPos;
+
+/**
+ * A lot inside the storage cell world that is assigned to a specific storage cell.
+ */
+public class StorageCellLot {
+
+    private static final int MAX_SIZE = 512;
+
+    /**
+     * Id of the lot.
+     */
+    private final int id;
+
+    /**
+     * The storage size of this dimension. This is dicateted by the pylon structure
+     * size used to perform the first transfer into this dimension. Once it's set,
+     * it cannot be changed anymore.
+     */
+    private final BlockPos size;
+
+    /**
+     * AE2 player id of who primarily owned the network when the storage lot was
+     * allocated.
+     */
+    private final int owner;
+
+    public StorageCellLot(int id, BlockPos size, int owner) {
+        this.id = id;
+        this.size = size;
+        this.owner = owner;
+        if (size.getX() < 1 || size.getY() < 1 || size.getZ() < 1) {
+            throw new IllegalArgumentException("Lot size " + size + " is smaller than minimum size.");
+        }
+        if (size.getX() > MAX_SIZE || size.getY() >= MAX_SIZE || size.getZ() >= MAX_SIZE) {
+            throw new IllegalArgumentException("Lot size " + size + " exceeds maximum size of " + MAX_SIZE);
+        }
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public BlockPos getSize() {
```

Comment:
Depending on absolute or relative values, the other one as additional helper might be worth it. Iirc some values use the relative size, while others like moving might benefit from having an easy access to the absolute limit.